parquet-variant/src/path.rs

```diff
@@ -43,10 +43,10 @@ use std::{borrow::Cow, ops::Deref};
 /// // access the field "foo" and then the first element in a variant list value
 /// let path = VariantPath::from("foo").join(0);
 /// // this is the same as the previous one
-/// let path2 = VariantPath::new(vec!["foo".into(), 0.into()]);
+/// let path2 = VariantPath::from_iter(["foo".into(), 0.into()]);
 /// assert_eq!(path, path2);
 /// // you can also create a path from a vector of `VariantPathElement` directly
-/// let path3 = VariantPath::new(vec![
+/// let path3 = VariantPath::from_iter([
 ///   VariantPathElement::field("foo"),
 ///   VariantPathElement::index(0)
 /// ]);
```

Comment:
Maybe we could show using the `collect` syntax too -- something like

```suggestion
/// let path3 = [
///   VariantPathElement::field("foo"),
///   VariantPathElement::index(0)
/// ].into_iter().collect::<VariantPath>()
```